packages/ra-ui-materialui/src/layout/UserMenu.tsx

```diff
@@ -23,14 +57,13 @@ export const UserMenu = (props: UserMenuProps) => {
         className,
         label = 'ra.auth.user_menu',
         icon = defaultIcon,
-        logout,
     } = props;
 
-    if (!logout && !children) return null;
-    const open = Boolean(anchorEl);
-
     const handleMenu = event => setAnchorEl(event.currentTarget);
-    const handleClose = () => setAnchorEl(null);
+    const handleClose = useCallback(() => setAnchorEl(null), []);
+    const context = useMemo(() => ({ onClose: handleClose }), [handleClose]);
+    if (!children) return null;
```

Comment:
By default, it should show the logout button.

Reply:
Unless a userMenu is provided, the `Appbar` pass its `logout` prop as the UserMenu only child (see https://github.com/marmelab/react-admin/pull/7214/files#diff-a81ad0901aeb0ca1ae0d7907215277afd74ce5e7c9691d1c7c67d1ca7d9952ccR100). Isn't it enoug?

Reply:
Maybe we could remove this `logout` prop alltogether btw

Reply:
If AppBar renders UserMenu with LogoutButton as child, it overrides the actual children of the user provided UserMenu. So I'm afraid we must only do that uf UserMenu doesn't have any child yet.

Reply:
> **Unless a userMenu is provided**

Reply:
You're right, I'm tired. Sorry for the noise.